rclpy/rclpy/node.py

```diff
@@ -1144,9 +1149,23 @@ def create_publisher(
 
         callback_group = callback_group or self.default_callback_group
 
+        failed = False
+        try:
+            final_topic = self.resolve_topic_name(topic)
+        except RuntimeError:
+            # if it's name validation error, raise a more appropriate exception.
+            self._validate_topic_or_service_name(topic)
```

Comment:
If we want to avoid seeing two exceptions, I think we can do the following (ref [PEP 409](https://www.python.org/dev/peps/pep-0409/)):

```suggestion
            try:
                self._validate_topic_or_service_name(topic)
            except InvalidTopicNameException as ex:
                 raise InvalidTopicNameException(ex) from None
```

Reply:
Done in https://github.com/ros2/rclpy/pull/635/commits/59951a392dc23461d3c0504a7179a3bfe1ddd88c.